### ISSUE TITLE
Select Options Button

### ISSUE THREAD

**Issue by DaveyR78** [holger1411/understrap#406]
Hi when clicking the select options button from the shop page - it adds the product to the basket without any options selected.
Should this not take you to the product page instead?
Thanks
Dave

**Comment by holger1411** [comment]
Could you post the URL?
Are the options "optional" or obligatory?

**Comment by DaveyR78** [comment]
Hi
try this site. https://www.doobysnails.com
they are set that you have to add at least one option to add it to the basket.
I can set up a test link if you need to look at code?
by the way great theme.
Thanks

**Comment by typeplus** [comment]
The customised **add-to-cart.php** chiefly the customised class **ajax_add_to_cart** is causing this. To fix you can honour the 3.0 methods of accessing product data as they are not public anymore by changing your **add-to-cart.php** to;

```
global $product;
echo apply_filters( 'woocommerce_loop_add_to_cart_link',
sprintf( '<div class="add-to-cart-container"><a href="%s" rel="nofollow" data-product_id="%s" data-product_sku="%s" data-quantity="%s" class="%s product_type_%s single_add_to_cart_button btn btn-outline-primary btn-block %s"> %s</a></div>',
esc_url( $product->add_to_cart_url() ),
esc_attr( $product->get_id() ),
esc_attr( $product->get_sku() ),
esc_attr( isset( $quantity ) ? $quantity : 1 ),
$product->is_purchasable() && $product->is_in_stock() ? 'add_to_cart_button' : '',
esc_attr( $product->get_type() ),
$product->get_type() == 'simple' ? 'ajax_add_to_cart' : '',
esc_html( $product->add_to_cart_text() )
),
$product );
```
